src/Infrastructure/Tests/Riganti.Utils.Infrastructure.EntityFramework.Tests/UnitOfWork/EntityFrameworkUnitOfWorkTests.cs

```diff
@@ -113,6 +184,74 @@ public async Task CommitAsync_CallCommitCoreOnlyIfHasOwnDbContext()
             unitOfWorkParentMock.Protected().Verify("CommitAsyncCore", Times.Once(), new CancellationToken());
         }
 
+        [Fact]
+        public async Task CommitAsync_ThrowIfChildCommitRequestedNotFulfilledByRoot()
+        {
+            Func<DbContext> dbContextFactory = () => new Mock<DbContext>().Object;
+
+            var unitOfWorkRegistryStub = new ThreadLocalUnitOfWorkRegistry();
```

Comment:
You should use `AsyncLocalUnitOfWorkRegistry` for async calls.

Reply:
ThreadLocalUnitOfWorkRegistry is used for consistency among tests.